src/main/groovy/edu/oregonstate/mist/recommendations/resources/RecommendationResource.groovy

```diff
@@ -0,0 +1,116 @@
+            @QueryParam('student_type') @NotEmpty String studentType,
+            @QueryParam('province') @NotEmpty String province,
+            @QueryParam('batch') @NotEmpty Integer batch,
+            @QueryParam('lower_limit') @NotEmpty Integer lowerLimit,
+            @QueryParam('upper_limit') @NotEmpty Integer upperLimit,
+            @QueryParam('year') Optional<Integer> year,
+            @QueryParam('major') Optional<String> major,
+            @QueryParam('language') Optional<String> language,
+            @QueryParam('page_size') Optional<Integer> pageSize,
+            @QueryParam('page_number') Optional<Integer> pageNum) {
+
+        List<Recommendation> recommendationList
+        studentType = translateStuType(studentType)
+        province = translateProvince(province)
+        if (by.toLowerCase() == "ranking") {
+            if (major.isPresent()) {
+                recommendationList = recommendationDAO.getMajorsByRank(
+                        province, studentType, batch, lowerLimit, upperLimit,
+                        year.orNull(), major.orNull(), pageSize.or(10), pageNum.or(1) - 1)
+
+            } else {
+                recommendationList = recommendationDAO.getUniversitiesByRank(
+                        province, studentType, batch, lowerLimit, upperLimit,
+                        year.orNull(), pageSize.or(10), pageNum.or(1) - 1)
+
+            }
+        } else if (by.toLowerCase() == "score-diff") {
+            // TO-DO
+            recommendationList = []
+        }
+        if (language.isPresent() && language.get() == "EN") {
+            recommendationList = translateResult (recommendationList)
+        }
+        recommendationList
+
+    }
+
+    /**
+     * Translate Student Type into Chinese if needed
+     * @param stuType
+     * @return Student Type In Chinese
+     */
+    private String translateStuType (String stuType) {
+        if (stuType.toLowerCase() == "arts") {
+            stuType = "文科"
+        } else if (stuType.toLowerCase() == "science") {
+            stuType = "理科"
+        }
+        stuType
+    }
+
+    /**
+     * Translate Province parameter into Chinese if needed
+     * @param province
+     * @return
+     */
+    private String translateProvince (String province)
+    {
+        String translate = Resource.properties.get(province)
+        if (translate != null && translate.length() > 0) {
+            translate
+        } else {
+            province
+        }
+    }
+
+    /**
+     *  Tranlate result into English
+     * @param recommendationList
+     * @return recommendationList
+     */
+    private List<Recommendation> translateResult (List<Recommendation> recommendationList) {
+
+        for (int i = 0; i < recommendationList.size(); i ++ ){
+            String universityName = recommendationList[i].getUniversity().name
+            String translate =  Resource.properties.get(universityName)
+            if (translate != null){
```

Comment:
It might be easier to have a small separate method called translate(String sourceText) that checks Resource.properties and checks the returned string. The check in this method is: translate != null and in the previous method you're also checking the lenght. Should you be trimming the string ? Having a separate method should help with this logic and keeping it all the same.